public/app/features/query/components/QueryGroup.tsx

```diff
@@ -111,7 +111,7 @@ export class QueryGroup extends PureComponent<Props, State> {
 
   onChangeDataSource = async (newSettings: DataSourceInstanceSettings) => {
     const { dsSettings } = this.state;
-    const queries = updateQueries(newSettings, this.state.queries, dsSettings);
+    const queries = await updateQueries(newSettings, this.state.queries, dsSettings);
 
     const dataSource = await this.dataSourceSrv.get(newSettings.name);
```

Comment:
updateQueries calls this too.  maybe we can pass this in?  or return it from updateQueries

Reply:
good catch!

Reply:
yea, passing it in and making updateQueries sync looks like a good idea

Reply:
or wait, it can't be sync as importFromAbstactQueries is async

Reply:
This is now fixed. As you noticed @torkelo the async nature of update queries is because of the abstract queries import.